packages/go_router/lib/src/go_route_information_provider.dart

```diff
@@ -0,0 +1,120 @@
+
+/// The route information provider created by go_router
+class GoRouteInformationProvider extends RouteInformationProvider
+    with WidgetsBindingObserver, ChangeNotifier {
+  /// Creates a [GoRouteInformationProvider].
+  GoRouteInformationProvider({
+    required RouteInformation initialRouteInformation,
+    Listenable? refreshListenable,
+  })  : _refreshListenable = refreshListenable,
+        _value = initialRouteInformation {
+    _refreshListenable?.addListener(notifyListeners);
+  }
+
+  final Listenable? _refreshListenable;
+
+  // ignore: unnecessary_non_null_assertion
+  static WidgetsBinding get _binding => WidgetsBinding.instance;
+
+  @override
+  void routerReportsNewRouteInformation(RouteInformation routeInformation,
+      {RouteInformationReportingType type =
+          RouteInformationReportingType.none}) {
+    // Avoid adding a new history entry if the route is the same as before.
+    final bool replace = type == RouteInformationReportingType.neglect ||
+        (type == RouteInformationReportingType.none &&
+            _valueInEngine.location == routeInformation.location);
+    SystemNavigator.selectMultiEntryHistory();
+    // TODO(chunhtai): should report extra to to browser through state if
+    // possible.
+    SystemNavigator.routeInformationUpdated(
+      location: routeInformation.location!,
+      replace: replace,
+    );
+    _value = routeInformation;
+    _valueInEngine = routeInformation;
+  }
+
+  @override
+  RouteInformation get value => DebugGoRouteInformation(
+        location: _value.location,
+        state: _value.state,
+      );
+  RouteInformation _value;
+  set value(RouteInformation other) {
+    final bool shouldNotify =
+        value.location != other.location || value.state != other.state;
+    _value = other;
+    if (shouldNotify) {
+      notifyListeners();
+    }
+  }
+
+  RouteInformation _valueInEngine =
+      RouteInformation(location: _binding.platformDispatcher.defaultRouteName);
+
+  void _platformReportsNewRouteInformation(RouteInformation routeInformation) {
+    if (_value == routeInformation) {
+      return;
+    }
+    _value = routeInformation;
+    _valueInEngine = routeInformation;
+    notifyListeners();
+  }
+
+  @override
+  void addListener(VoidCallback listener) {
+    if (!hasListeners) {
+      _binding.addObserver(this);
+    }
+    super.addListener(listener);
+  }
+
+  @override
+  void removeListener(VoidCallback listener) {
+    super.removeListener(listener);
+    if (!hasListeners) {
+      _binding.removeObserver(this);
+    }
+  }
+
+  @override
+  void dispose() {
+    if (hasListeners) {
+      _binding.removeObserver(this);
+    }
+    _refreshListenable?.removeListener(notifyListeners);
+    super.dispose();
+  }
+
+  @override
+  Future<bool> didPushRouteInformation(
+      RouteInformation routeInformation) async {
+    assert(hasListeners);
+    print('_platformReportsNewRouteInformation $routeInformation');
```

Comment:
Is this print intentional? Or should it be wrapped in an assert?

Reply:
Oops, I'll take this out. So far, we've been using `package:logging` to log messages intended for users.